File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int n = numbers.size();
        int start = 0, end = n - 1;
        vector<int>ans(2,0);
        while(start < end)
        {
            int sum = numbers[start] + numbers[end];
            if(sum > target)
                end--;
            else if(sum == target)
            {
                ans[0] = start+1;
                ans[1] = end+1;
                break;
            }
            else
                start++;
        }
        return ans;
    }
};

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        #hashmap store indices -> 2 loops
        #hashmap store indcies -> 1 loop
        start, end = 0, len(numbers)-1
        while start < end:
            sumNum = numbers[start] + numbers[end]
            if sumNum == target:
                return [start+1, end+1]
            elif sumNum > target:
                end-=1
            else:
                start+=1
        return []
            